uisdk/src/main/java/com/karhoo/uisdk/screen/booking/quotes/filterview/ServiceAgreementsFilter.kt

```diff
@@ -9,17 +9,30 @@ class ServiceAgreementsFilter(selectedTypes: ArrayList<MultiSelectData>) :
         if (selectedTypes.size == 0) {
             return true
         }
-        return (quote.serviceAgreements?.freeCancellation != null && selectedTypes.any {
-            it.fixedTag?.contains(
-                FREE_CANCELLATION_TAG
-            ) == true
-        })
-                ||
-                (quote.serviceAgreements?.freeWaitingTime != null && selectedTypes.any {
-                    it.fixedTag?.contains(
-                        FREE_WAITING_TIME_TAG
-                    ) == true
-                })
+        if (selectedTypes.size == 1)
+            return (quote.serviceAgreements?.freeCancellation != null && selectedTypes.any {
+                it.fixedTag?.contains(
+                    FREE_CANCELLATION_TAG
+                ) == true
```

Comment:
Think we can put each one of these in a function () : Boolean to simplify the sintax

Something like 
quoteContainsTag(quote, selectedTypes, tag): Boolean { 
 return (quote.serviceAgreements?.freeWaitingTime != null && selectedTypes.any {
                        it.fixedTag?.contains(
                            tag
                        ) == true
                    })
}

And then replace the paragrahps from 12 to 35